plugins/janus_textroom.c

```diff
@@ -1453,9 +1453,9 @@ janus_plugin_result *janus_textroom_handle_incoming_request(janus_plugin_session
 			g_snprintf(error_cause, 512, "No such room (%s)", room_id_str);
 			goto msg_response;
 		}
+		janus_mutex_unlock(&rooms_mutex);
```

Comment:
Better keep the `janus_refcount_increase` call within the lock.

Reply:
I've updated the PR to take into consideration all of your comments.